api/cpp/include/slint-platform.h

```diff
@@ -158,6 +166,19 @@ class Platform
     }
 #endif
 
+    /// Sends the given text into the system clipboard.
+    ///
+    /// If the platform doesn't support the specified clipboard, this function should do nothing
+    virtual void set_clipboard_text(const std::string_view &text, Clipboard clipboard) { }
+
+    /// Returns a copy of text stored in the system clipboard, if any.
+    ///
+    /// If the platform doesn't support the specified clipboard, the function should return nullopt
+    virtual std::optional<std::string_view> clipboard_text(Clipboard clipboard)
```

Comment:
API wise I'm uncertain returning a `string_view` is a wise choice here. This implies that the sub-class must basically keep a copy of the clipboard text as a member variable, for it to be safe. I think it would be better to return an optional slint::SharedString. But since the Rust API returns also a `String` instead of a `SharedString`, I guess it would really have to be either a `std::string` or a `std::u8string` ?

Reply:
Moved to shared string as of e1f553d
I'm curious about std::u8string though. can that be converted to a `String` without doing a copy? I believe atm there is a copy in `String::from(out_text)`